lib/array_association_scope.rb

```diff
@@ -4,13 +4,41 @@ def self.scope(association, connection)
     INSTANCE.scope association, connection
   end
 
+  class BindSubstitution
+    def initialize(block)
+      @block = block
+    end
+
+    def bind_value(scope, column, value, alias_tracker)
+      substitute = alias_tracker.connection.substitute_at(column)
+      cast_type = ActiveRecord::ConnectionAdapters::PostgreSQL::OID::Array
```

Comment:
I'm wondering why there was a need to create a new Array type object when *I assume* the column that's being bound is an array already.